### ISSUE TITLE
Trying to implement a custom dataset

### ISSUE THREAD

**Issue by sceddd** [ultralytics/yolov5#13067]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I'm trying to run a custom dataset, my yaml file look like this:
train:
  - ../YOLOv5-Lite/data/dataset/images/train/
val:
  - ../YOLOv5-Lite/data/dataset/images/val/

nc: 10

names:
  0: person
  1: car
  2: cat
  3: dog
  4: chair
  5: motorbike
  6: bird
  7: bicycle
  8: bus
  9: sheep
  
and my folder look like this:
![image](https://github.com/ultralytics/yolov5/assets/102157918/52968fa2-dded-4906-993c-85ae25887936)
I have tried to use roboflow also to create one, and use it but it still pop up this kind of error:
![image](https://github.com/ultralytics/yolov5/assets/102157918/3a0d45e8-e3e8-41de-874a-5c6cebe3ab0d)
Hope some one could help

**Comment by glenn-jocher** [comment]
Hello! It looks like you're on the right track with setting up your custom dataset. From the YAML configuration and folder structure you've shared, everything seems correctly formatted at first glance. However, the error screenshots are not visible here, which makes it a bit challenging to diagnose the issue precisely.

A common issue could be related to the paths specified in your YAML file or the format of the images and labels. Ensure that the paths are correct relative to the location from which you are running the training script. Also, verify that your images and corresponding label files are correctly paired and formatted as expected by YOLOv5.

If you're still facing issues, could you please provide the exact error message you're encountering? This will help in pinpointing the problem more effectively.

For a detailed step-by-step guide on preparing your dataset, you might want to revisit our documentation on training with custom data [here](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/). This guide covers everything from image collection to labeling and dataset preparation.

Looking forward to helping you resolve this! 🚀

**Comment by sceddd** [comment]
By any chance, it's because of the annotation file, for example:

0 0.6683991683991685 0.6156250000000001 0.6465696465696467 0.728125

0 is the person type, but there is no class 0 in the yaml file?
because I saw this log in the output also:
![image](https://github.com/ultralytics/yolov5/assets/102157918/9e32132e-0e12-4256-bda2-3579be57f06d)
well you can see 1552 missing in the log, and total file in my val file is 1552.

**Comment by sceddd** [comment]
One more piece of information, that I'm using a model name's YOLOv5-Lite, so maybe It's not because of this repo but I just wanna make sure that my folder are all correct so that I can find another solution for it.

**Comment by glenn-jocher** [comment]
@sceddd hello!

It sounds like you're using a variant of YOLOv5, specifically YOLOv5-Lite, which might not be directly supported in the standard YOLOv5 repository. However, ensuring your dataset is correctly formatted is crucial regardless of the model variant.

From your description, it seems that the class index in your annotation files might not be matching the classes defined in your YAML file. The class index `0` in your annotation should correspond to the first class listed in your YAML file. If there's a mismatch or if the class indices are not zero-indexed starting from `0`, this could cause issues during training.

Also, the log indicating missing labels for all images in your validation set suggests that the labels might not be correctly located or formatted. Double-check that your label files are in the correct directory and that their format matches the expected YOLO format (`class x_center y_center width height` with normalized values).

If everything seems correct with your dataset and you continue to experience issues, it might be worth checking specifics related to YOLOv5-Lite or reaching out to the maintainers of that specific repository for more tailored support.

Best of luck with your model training! 🚀
